drivers/spi/spi_xec_qmspi_ldma.c

```diff
@@ -159,36 +162,42 @@ static void qmspi_reset(struct qmspi_regs *regs)
 	regs->TM_TAPS_CTRL = taps[2];
 }
 
-/*
- * Calculate 8-bit QMSPI frequency divider register field value.
- * QMSPI input clock is MCHP_QMSPI_INPUT_CLOCK_FREQ_HZ.
- * The 8-bit divider is encoded as:
- * 0 is divide by 256
- * 1 through 255 is divide by this value.
- */
-static uint32_t qmspi_encoded_fdiv(uint32_t freq_hz)
+static uint32_t qmspi_encoded_fdiv(const struct device *dev, uint32_t freq_hz)
 {
-	uint32_t fdiv = 1;
+	struct spi_qmspi_data *qdata = dev->data;
 
-	if (freq_hz < (MCHP_QMSPI_INPUT_CLOCK_FREQ_HZ / 256u)) {
-		fdiv = 0; /* HW fdiv of 0 -> divide by 256 */
-	} else if (freq_hz < MCHP_QMSPI_INPUT_CLOCK_FREQ_HZ) {
-		fdiv = MCHP_QMSPI_INPUT_CLOCK_FREQ_HZ / freq_hz;
+	if (freq_hz == 0u) {
+		return 0u; /* maximum frequency divider */
 	}
 
-	return fdiv;
+	return (qdata->base_freq_hz / freq_hz);
 }
 
-/* Program QMSPI frequency divider field in mode register */
-static void qmspi_set_frequency(struct qmspi_regs *regs, uint32_t freq_hz)
+/* Program QMSPI frequency divider field in the mode register.
+ * MEC172x QMSPI input clock source is the Fast Peripheral domain whose
+ * clock is controlled by the PCR turbo clock. 96 MHz if turbo mode
+ * enabled else 48 MHz. Query the clock control driver to get clock
+ * rate of fast peripheral domain. MEC172x QMSPI clock divider has
+ * been expanded to a 16-bit field encoded as:
+ * 0 = divide by 0x10000
+ * 1 to 0xffff = divide by this value.
+ */
+static int qmspi_set_frequency(struct qmspi_regs *regs, uint32_t freq_hz)
 {
-	uint32_t fdiv, qmode;
+	clock_control_subsys_t clkss =
+		(clock_control_subsys_t)(MCHP_XEC_PCR_CLK_PERIPH_FAST);
+	uint32_t clock = MCHP_QMSPI_INPUT_CLOCK_FREQ_HZ;
```

Comment:
Rename variable "clock" to fix check violation

Reply:
ok